sdk/digitaltwins/azure-digitaltwins-core/src/main/java/com/azure/digitaltwins/core/DigitalTwinsAsyncClient.java

```diff
@@ -228,4 +234,126 @@ public HttpPipeline getHttpPipeline() {
+                return Mono.error(e);
+            }
+        }
+
+        return protocolLayer.getDigitalTwinModels().addWithResponseAsync(modelsPayload, context)
+            .map(
+                listResponse -> new PagedResponseBase<>(
+                    listResponse.getRequest(),
+                    listResponse.getStatusCode(),
+                    listResponse.getHeaders(),
+                    listResponse.getValue(),
+                    null,
+                    ((ResponseBase)listResponse).getDeserializedHeaders()));
+    }
+
+    /**
+     * Gets a model, including the model metadata and the model definition.
+     * @param modelId The Id of the model.
+     * @return The application/json model
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<ModelData> getModel(String modelId) {
+        return withContext(context -> getModel(modelId, context));
+    }
+
+    Mono<ModelData> getModel(String modelId, Context context){
+        return protocolLayer.getDigitalTwinModels().getByIdWithResponseAsync(modelId, includeModelDefinition, context)
+            .flatMap(modelDataResponse -> Mono.just(mapper.convertValue(modelDataResponse.getValue(), ModelData.class)));
+    }
+
+    /**
+     * Gets a model, including the model metadata and the model definition.
+     * @param modelId The Id of the model.
+     * @return The application/json model
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<ModelData>> getModelWithResponse(String modelId) {
+        return withContext(context -> getModelWithResponse(modelId, context));
+    }
+
+    Mono<Response<ModelData>> getModelWithResponse(String modelId, Context context){
+        return protocolLayer
+            .getDigitalTwinModels()
+            .getByIdWithResponseAsync(modelId, includeModelDefinition, context);
+    }
+
+    /**
+     * Gets the list of models by iterating through a collection.
+     * @param listModelOptions The options for the list operation.
+     * @return A {@link PagedFlux} of ModelData.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<ModelData> listModels(ListModelOptions listModelOptions) {
+        return new PagedFlux<>(
+            () -> withContext(context -> listModelsSinglePageAsync(listModelOptions, context)),
+            nextLink -> withContext(context -> listModelsNextSinglePageAsync(nextLink, context)));
+    }
+
+    PagedFlux<ModelData> listModels(ListModelOptions listModelOptions, Context context){
+        return new PagedFlux<>(
+            () -> listModelsSinglePageAsync(listModelOptions, context),
+            nextLink -> listModelsNextSinglePageAsync(nextLink, context));
+    }
+
+    Mono<PagedResponse<ModelData>> listModelsSinglePageAsync(ListModelOptions listModelOptions, Context context){
+        return protocolLayer.getDigitalTwinModels().listSinglePageAsync(
+            (List<String>) listModelOptions.getDependenciesFor(),
```

Comment:
Why not define `listModelOptions.getDependenciesFor` as a List to begin with? If the PL is restrictive, then any advantage in the public API being flexible? 
Is there a risk of running into conversion errors by simply casting it to a List -> would a for-each loop be better: https://www.baeldung.com/java-iterable-to-collection?

Reply:
I think so too, I will change all the types back to List since that is the input the protocol layer accepts and I don't see a point in going more generic than that